tests/lib/azure/vnet.go

```diff
@@ -32,8 +32,10 @@ func (vnet *Vnet) AssertExists(t *testing.T, name string, address string) {
 		}
 
 		addressActual := *net.VirtualNetworkPropertiesFormat.AddressSpace.AddressPrefixes
-		if addressActual[0] != address {
-			return errors.New("Address expected is " + address + " but actual is " + addressActual[0])
+		if addressActual != nil {
+			if addressActual[0] != address {
+				return errors.New("Address expected is " + address + " but actual is " + addressActual[0])
+			}
```

Comment:
I think here you should return an error if the field is nil. Because if you ignore it the test will pass. 
Would be nice to check for the other fields too, like AddressSpace. :)